src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -163,12 +162,19 @@ case class WorkflowActor(workflow: WorkflowDescriptor,
       stay()
   }
 
-  // FSM will call *all* onTransition handlers which are defined for a particular state transition.
-  // This handler will update workflow state for all transitions.
+  /*
+    FSM will call *all* onTransition handlers which are defined for a particular state transition.
+    This handler will update workflow state for all transitions.
+  */
   onTransition {
     case fromState -> toState =>
       log.info(s"$tag transitioning from $fromState to $toState.")
       dataAccess.updateWorkflowState(workflow.id, toState)
+      /*
+        Send a message to self to trigger an actor shutdown. Run on a short timer to help enable some
+        unit test instrumentation
+       */
+      if (toState.isTerminal) setTimer("terminate with extreme prejudice", Terminate, AkkaTimeout, DontRepeatTimer)
```

Comment:
While amusing, when this message crops up in the logs it might cause some undue consternation.